src/main/java/oracle/kubernetes/operator/builders/WatchBuilder.java

```diff
@@ -0,0 +1,320 @@
+            }
+        }
+    }
+
+    /**
+     * Creates a web hook object to track pods
+     * @return the active web hook
+     * @throws ApiException if there is an error on the call that sets up the web hook.
+     */
+    public Watch<V1Pod> createPodWatch(String namespace) throws ApiException {
+        return FACTORY.createWatch(clientHolder, callParams, V1Pod.class, new ListPodCall(namespace));
+    }
+
+    private class ListPodCall implements BiFunction<ClientHolder, CallParams, Call> {
+        private String namespace;
+
+        ListPodCall(String namespace) {
+            this.namespace = namespace;
+        }
+
+        @Override
+        public Call apply(ClientHolder clientHolder, CallParams callParams) {
+            try {
+                return clientHolder.getCoreApiClient().listNamespacedPodCall(namespace, callParams.getPretty(),
+                            START_LIST, callParams.getFieldSelector(), callParams.getIncludeUninitialized(),
+                            callParams.getLabelSelector(), callParams.getLimit(), callParams.getResourceVersion(),
+                            callParams.getTimeoutSeconds(), WATCH, null, null);
+            } catch (ApiException e) {
+                throw new UncheckedApiException(e);
+            }
+        }
+    }
+
+    /**
+     * Creates a web hook object to track changes to the cluster ingress
+     * @return the active web hook
+     * @throws ApiException if there is an error on the call that sets up the web hook.
+     */
+    public Watch<V1beta1Ingress> createIngressWatch(String namespace) throws ApiException {
+        return FACTORY.createWatch(clientHolder, callParams, V1beta1Ingress.class, new ListIngressCall(namespace));
+    }
+
+    private class ListIngressCall implements BiFunction<ClientHolder, CallParams, Call> {
+        private String namespace;
+
+        ListIngressCall(String namespace) {
+            this.namespace = namespace;
+        }
+
+        @Override
+        public Call apply(ClientHolder clientHolder, CallParams callParams) {
+            try {
+                return clientHolder.getExtensionsV1beta1ApiClient().listNamespacedIngressCall(namespace,
+                            callParams.getPretty(), START_LIST, callParams.getFieldSelector(),
+                            callParams.getIncludeUninitialized(), callParams.getLabelSelector(), callParams.getLimit(),
+                            callParams.getResourceVersion(), callParams.getTimeoutSeconds(), WATCH, null, null);
+            } catch (ApiException e) {
+                throw new UncheckedApiException(e);
+            }
+        }
+    }
+
+    /**
+     * Creates a web hook object to track changes to weblogic domains in all namespaces
+     * @return the active web hook
+     * @throws ApiException if there is an error on the call that sets up the web hook.
+     */
+    public Watch<Domain> createDomainsInAllNamespacesWatch() throws ApiException {
```

Comment:
These "All" namespace variants will only be used by test code.  Runtime code will always work with some set of namespaces that will require separate watches for each namespace.